presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/reloading/ReloadingResourceGroupConfigurationManager.java

```diff
@@ -220,6 +223,12 @@ public synchronized void load()
             if (lastRefresh.get() != 0) {
                 log.debug("Last successful configuration loading was %s ago", succinctNanos(System.nanoTime() - lastRefresh.get()).toString());
             }
+
+            if(failFastOnValidation)
+            {
+                log.error(e, "Fail fast mode is enabled. Failing the loading operation");
+                throw e;
+            }
```

Comment:
1. Will this result in query failures immediately? I think it won't as this will just fail the reloading logic which is not in part of the query flow.
2. Can we achieve failing fast by just reducing the timeout maxRefreshInterval?

Reply:
I think you are right, we should be able to just tune the maxRefreshInterval